python/lsst/ts/externalscripts/base_parameter_march.py

```diff
@@ -0,0 +1,461 @@
+                type: string
+                default: PARAMETER_MARCH
+              reason:
+                description: Optional reason for taking the data.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              ignore:
+                description: >-
+                    CSCs from the group to ignore in status check. Name must
+                    match those in self.group.components, e.g.; hexapod_1.
+                type: array
+                items:
+                  type: string
+            oneOf:
+                - required:
+                    - dofs
+                    - range
+                    - n_steps
+                - required:
+                    - dofs
+                    - step_sequence
+                - required:
+                    - dof_index
+                    - range
+                    - n_steps
+                - required:
+                    - dof_index
+                    - step_sequence
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super().get_schema()
+
+        for properties in base_schema_dict["properties"]:
+            schema_dict["properties"][properties] = base_schema_dict["properties"][
+                properties
+            ]
+
+        return schema_dict
+
+    async def configure(self, config: types.SimpleNamespace) -> None:
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        await self.configure_tcs()
+        await self.configure_camera()
+        await self.configure_ocps()
+
+        if hasattr(config, "ignore"):
+            for comp in config.ignore:
+                if comp in self.tcs.components_attr:
+                    self.log.debug(f"Ignoring TCS component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+                elif comp in self.camera.components_attr:
+                    self.log.debug(f"Ignoring Camera component {comp}.")
+                    setattr(self.camera.check, comp, False)
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Groups. "
+                        f"Must be one of {self.tcs.components_attr} or "
+                        f"{self.camera.components_attr}. Ignoring."
+                    )
+
+        if hasattr(config, "step_sequence"):
+            config.range = config.step_sequence[-1] - config.step_sequence[0]
+            config.n_steps = len(config.step_sequence)
+        elif hasattr(config, "range"):
+            config.step_sequence = np.linspace(
+                -config.range, config.range, config.n_steps
+            ).tolist()
+
+        if hasattr(config, "rotation_sequence"):
```

Comment:
same comment as above. I would rather you leave `config` object alone and create class attributes to hold what you need here.